src/proof/nova.rs

```diff
@@ -678,3475 +520,163 @@ where
-        let min_len = a.len().min(b.len());
-        for i in 0..min_len {
-            if a[i] != b[i] {
-                return Some((i, (Some(a[i]), Some(b[i]))));
-            }
-        }
-        match (a.get(min_len), b.get(min_len)) {
-            (Some(&a_elem), None) => Some((min_len, (Some(a_elem), None))),
-            (None, Some(&b_elem)) => Some((min_len, (None, Some(b_elem)))),
-            _ => None,
-        }
-    }
-
-    pub fn test_aux<'a, F: CurveCycleEquipped, C: Coprocessor<F>, M: MultiFrameTrait<'a, F, C>>(
-        s: &'a M::Store,
-        expr: &str,
-        expected_result: Option<M::Ptr>,
-        expected_env: Option<M::Ptr>,
-        expected_cont: Option<M::ContPtr>,
-        expected_emitted: Option<&[M::Ptr]>,
-        expected_iterations: usize,
-        lang: Option<Arc<Lang<F, C>>>,
-    )
-    // technical bounds that would disappear once associated_type_bounds stabilizes
-    where
-        <<G1<F> as Group>::Scalar as ff::PrimeField>::Repr: Abomonation,
-        <<G2<F> as Group>::Scalar as ff::PrimeField>::Repr: Abomonation,
-    {
-        for chunk_size in REDUCTION_COUNTS_TO_TEST {
-            nova_test_full_aux::<F, C, M>(
-                s,
-                expr,
-                expected_result,
-                expected_env,
-                expected_cont,
-                expected_emitted,
-                expected_iterations,
-                chunk_size,
-                false,
-                None,
-                lang.clone(),
-            )
-        }
-    }
-
-    fn nova_test_full_aux<
-        'a,
-        F: CurveCycleEquipped,
-        C: Coprocessor<F>,
-        M: MultiFrameTrait<'a, F, C>,
-    >(
-        s: &'a M::Store,
-        expr: &str,
-        expected_result: Option<M::Ptr>,
-        expected_env: Option<M::Ptr>,
-        expected_cont: Option<M::ContPtr>,
-        expected_emitted: Option<&[M::Ptr]>,
-        expected_iterations: usize,
-        reduction_count: usize,
-        check_nova: bool,
-        limit: Option<usize>,
-        lang: Option<Arc<Lang<F, C>>>,
-    )
-    // technical bounds that would disappear once associated_type_bounds stabilizes
-    where
-        <<G1<F> as Group>::Scalar as ff::PrimeField>::Repr: Abomonation,
-        <<G2<F> as Group>::Scalar as ff::PrimeField>::Repr: Abomonation,
-    {
-        let expr = s.read(expr).unwrap();
-
-        let f = |l| {
-            nova_test_full_aux2::<F, C, M>(
-                s,
-                expr,
-                expected_result,
-                expected_env,
-                expected_cont,
-                expected_emitted,
-                expected_iterations,
-                reduction_count,
-                check_nova,
-                limit,
-                l,
-            )
-        };
-
-        if let Some(l) = lang {
-            f(l)
-        } else {
-            let lang = Lang::new();
-            f(Arc::new(lang))
-        };
+// TODO Move to `circuit/circuit_frame.rs`
```

Comment:
Nit: I'm not convinced this is necessary, specifically because this is inherently very tied to Nova. I would remove the comment, and redirect the thinking behind it to ditching SnarkPack. Once we do take *that* decision, then pushing this sort of nova-specific impl across the code base makes more sense to me.